nova/src/utils.rs

```diff
@@ -35,15 +35,21 @@ pub fn iter_bits_le(bytes: &[u8]) -> impl Iterator<Item = bool> + '_ {
         .flat_map(|byte| (0..8).map(move |bit| ((1 << bit) & byte) != 0))
 }
 
-/// An implementation of base_2 log that will never fail, instead returning 0 for all values smaller than 1.
+/// An implementation of (almost) base_2 integer log that will never fail.
+///
+/// Returns:
+///   log_2(x) = y for x > 1
+///              1 for x = 1
+///              0 otherwise
+///
 /// It is 'safe' in the sense that it will not panic, unlike just calling `.unwrap()` on a `.checked_ilog2`.
-/// Usually used to compute the number of bits needed to index a set or a vector.
+/// Used to compute the number of bits needed to index a set or a vector.
 #[macro_export]
-macro_rules! safe_log {
+macro_rules! safe_loglike {
```

Comment:
`ceil_log2` maybe?

Reply:
It's not quite `ceil` though, and neither is it `clamped` because `log_2 = 0` and we're redefining that.